src/ucs/datastruct/array.c

```diff
@@ -47,7 +48,17 @@ ucs_status_t ucs_array_grow(void **buffer_p, size_t *capacity_p,
     memcpy(new_buffer, *buffer_p, current_capacity * value_size);
 
 out:
+    if (old_buffer_p == NULL) {
+        ucs_array_free(old_buffer);
```

Comment:
maybe `ucs_array_buffer_free()` with actual function since `ucs_array_{grow,resize,...}()` operate on the `ucs_array_s()` itself?

Reply:
did not understand the comment.. can you pls elaborate?

Reply:
actually i now see that `ucs_array_free()` is already a function below. minor: regarding naming, since `ucs_array_{append,last,reserve}()` all operate on `ucs_array_s(<type>)`, i would expect we keep `ucs_array_free` naming if we need to oerate/free `ucs_array_s()` object.

Reply:
yes ;)